x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -480,6 +486,52 @@ private PhysicalOperation planEnrich(EnrichExec enrich, LocalExecutionPlannerCon
         );
     }
 
+    private PhysicalOperation planHashJoin(HashJoinExec join, LocalExecutionPlannerContext context) {
+        PhysicalOperation source = plan(join.child(), context);
+        int positionsChannel = source.layout.numberOfChannels();
+
+        Layout.Builder layoutBuilder = source.layout.builder();
+        for (Attribute f : join.output()) {
+            if (join.child().outputSet().contains(f)) {
+                continue;
+            }
+            layoutBuilder.append(f);
```

Comment:
Thought: Interestingly, this operator seems to only add columns/channels/attributes, even in case of name collisions. If that's correct, there's opportunity to reduce memory usage by projecting out shadowed attributes.

This applies similarly to e.g. `planGrok`, which also only ever adds new attributes to the layout, even in case of shadowing.

Reply:
I'm skipping this one for the moment. Maybe a follow up.